pallets/capacity/src/benchmarking.rs

```diff
@@ -1,17 +1,42 @@
 use super::*;
 use crate::Pallet as Capacity;
 
-use frame_benchmarking::{benchmarks, whitelisted_caller};
+use frame_benchmarking::{account, benchmarks, whitelist_account, Vec};
+use frame_support::{assert_ok, log::info, traits::Currency};
 use frame_system::RawOrigin;
 
+const SEED: u32 = 0;
+
+pub fn register_provider<T: Config>(target_id: MessageSourceId, name: &'static str) {
+	let name = Vec::from(name).try_into().expect("error");
+	assert_ok!(T::BenchmarkHelper::create(target_id, name));
+}
+pub fn create_funded_account<T: Config>(
+	string: &'static str,
+	n: u32,
+	balance_factor: u32,
+) -> T::AccountId {
+	let user = account(string, n, SEED);
+	whitelist_account!(user);
+	let balance = T::Currency::minimum_balance() * balance_factor.into();
+	let _ = T::Currency::make_free_balance_be(&user, balance);
+	assert_eq!(T::Currency::free_balance(&user), balance.into());
+	user
+}
+
 benchmarks! {
 	stake {
-		let caller: T::AccountId = whitelisted_caller();
-		let amount: BalanceOf<T> = 1u32.into();
+		let caller: T::AccountId = create_funded_account::<T>("account", SEED, 5u32);
+		let amount: BalanceOf<T> = T::MinimumStakingAmount::get();
 		let target = 1;
 
+		register_provider::<T>(target, "Foo");
+
 	}: _ (RawOrigin::Signed(caller.clone()), target, amount)
 	verify {
+		assert!(StakingAccountLedger::<T>::contains_key(&caller));
+		assert!(StakingTargetLedger::<T>::contains_key(&caller, target));
+		assert!(CapacityLedger::<T>::contains_key(target));
```

Comment:
did you also want to check for any events here?

Reply:
Ahh, good idea.